### DISCUSSION TITLE
html pure js help

### DISCUSSION THREAD

**Discussion by Rickshanelorie** [WalletConnect/walletconnect-monorepo#3271] [Developer Support]
i want to get the address for this wallet connect code pure js and html help  import {
  EthereumClient,
  w3mConnectors,
  w3mProvider,
  WagmiCore,
  WagmiCoreChains,
  WagmiCoreConnectors
} from "https://unpkg.com/@web3modal/ethereum@2.6.2";
import { Web3Modal } from "https://unpkg.com/@web3modal/html@2.6.2";

// 0. Import wagmi dependencies
const { bscTestnet, bsc } = WagmiCoreChains;
const { configureChains, createConfig } = WagmiCore;

// 1. Define chains
const chains = [ bscTestnet, bsc ];
const projectId = "b47b978a9df1c500b5f865a4c5e7c3dc";

// 2. Configure wagmi client
const { publicClient } = configureChains(chains, [w3mProvider({ projectId })]);
const wagmiConfig = createConfig({
  autoConnect: true,
  connectors: [
    ...w3mConnectors({ chains, version: 2, projectId }),
    new WagmiCoreConnectors.CoinbaseWalletConnector({
      chains,
      options: {
        appName: "html wagmi example",
      },
    }),
  ],
  publicClient,
});

// 3. Create ethereum and modal clients
const ethereumClient = new EthereumClient(wagmiConfig, chains);
export const web3Modal = new Web3Modal(
  {
    projectId,
    walletImages: {
      safe: "https://pbs.twimg.com/profile_images/1566773491764023297/IvmCdGnM_400x400.jpg",
    },
  },
  ethereumClient
);

**Comment by glitch-txs** [comment]
you can use wagmiCore to call getAccount function
https://wagmi.sh/core/actions/getAccount

**Reply by Rickshanelorie** [reply]
how can i use that with html it don't have an cdn so can you show me ?? how to do it?

**Reply by Rickshanelorie** [reply]
would highly appreciate it if you could help me here

**Reply by glitch-txs** [reply]
same way as shown in the code you share, you can try console logging `WagmiCore` to check all functions available.

`const { getAccount } = WagmiCore`

for the rest you can follow wagmi's documentation :)

**Reply by Rickshanelorie** [reply]
here's my updates import {
  EthereumClient,
  w3mConnectors,
  w3mProvider,
  WagmiCore,
  WagmiCoreChains,
  WagmiCoreConnectors
} from "https://unpkg.com/@web3modal/ethereum@2.6.2";
import { Web3Modal } from "https://unpkg.com/@web3modal/html@2.6.2";

// 0. Import wagmi dependencies
const { bscTestnet, bsc } = WagmiCoreChains;
const { configureChains, createConfig } = WagmiCore;

// 1. Define chains
const chains = [ bscTestnet, bsc ];
const projectId = "b47b978a9df1c500b5f865a4c5e7c3dc";

// 2. Configure wagmi client
const { publicClient } = configureChains(chains, [w3mProvider({ projectId })]);
const wagmiConfig = createConfig({
  autoConnect: true,
  connectors: [
    ...w3mConnectors({ chains, version: 2, projectId }),
    new WagmiCoreConnectors.CoinbaseWalletConnector({
      chains,
      options: {
        appName: "html wagmi example",
      },
    }),
  ],
  publicClient,
});

// 3. Create ethereum and modal clients
const ethereumClient = new EthereumClient(wagmiConfig, chains);
export const web3Modal = new Web3Modal(
  {
    projectId,
    walletImages: {
      safe: "https://pbs.twimg.com/profile_images/1566773491764023297/IvmCdGnM_400x400.jpg",
    },
  },
  ethereumClient
);

document.getElementById('my-button').addEventListener('click', async () => {
  try {
    const accounts = await ethereum.request({ method: "eth_requestAccounts" });
    const address = accounts[0];
    web3.eth.setProvider(Web3.givenProvider);
    
    // Convert the address to a checksum address
    let _wallet = web3.utils.toChecksumAddress(address);
    
    // Modify the address to display only the first 3 characters and the last 4 characters
    const modifiedAddress = _wallet.slice(0, 3) + "..." + _wallet.slice(_wallet.length - 4, _wallet.length);
    
    console.log("User's Ethereum address: ", modifiedAddress);
  } catch (e) {
    console.error("Failed to request Ethereum address: ", e);
  }
  web3Modal.openModal();
});

**Comment by Rickshanelorie** [comment]
still don't get it from the docs after i call the getAccount how can do the same as this ethereum
    .request({ method: "eth_requestAccounts" })
    .then(async (accounts) => {
      address = accounts[0];
      web3.eth.setProvider(Web3.givenProvider);
      let _wallet = web3.utils.toChecksumAddress(address);
